sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativeCrossVersion.scala

```diff
@@ -5,38 +5,21 @@ package sbtplugin
 
 import sbt._
 import scala.scalanative.nir.Versions
+import SBTCompat._
 
 object ScalaNativeCrossVersion {
+  private final val ReleaseVersion =
+    raw"""(\d+)\.(\d+)\.(\d+)""".r
 
-  val currentBinaryVersion = {
-    val ReleaseVersion = raw"""(\d+)\.(\d+)\.(\d+)""".r
+  val currentBinaryVersion = binaryVersion(Versions.current)
 
-    def binaryVersion(full: String): String = full match {
-      case ReleaseVersion(major, minor, _) => s"$major.$minor"
-      case _                               => full
-    }
-
-    binaryVersion(Versions.current)
+  def binaryVersion(full: String): String = full match {
+    case ReleaseVersion(major, _, _) => major
```

Comment:
This change would have been devastating if it made into a release. It turned `native0.3` binary version into `native0`. #1096

Reply:
nice catch!